cloud/pkg/cloudstream/streamserver.go

```diff
@@ -106,7 +109,23 @@ func (s *StreamServer) getContainerLogs(r *restful.Request, w *restful.Response)
 		}
 	}()
 
-	sessionKey := strings.Split(r.Request.Host, ":")[0]
+	// extract pod namespace and pod name from request
+	meta := strings.Split(r.Request.URL.Path, "/")
+	namespaceName := meta[2]
+	podName := meta[3]
+
+	kubeClient := client.GetKubeClient()
+	if kubeClient == nil {
+		klog.Fatalf("cannot get kube client\n")
+		return
+	}
+
+	pod, err := kubeClient.CoreV1().Pods(namespaceName).Get(context.Background(), podName, v1.GetOptions{})
+	if err != nil {
+		klog.Errorf("get pod %s/%s failed: %v\n", namespaceName, podName, err)
+		return
+	}
+	sessionKey := pod.Spec.NodeName
 	session, ok := s.tunnel.getSession(sessionKey)
```

Comment:
when use nodeName, IMO, we also need to use nodeName as key to store in the tunnel map cache.

Reply:
Yes you are right, but the nodeName is similiar to the config parameter `hostnameOveride` from the edge and this will already be used in the tunnel map.

Reply:
as @subpathdev mentioned the hostname and the IP are already used as keys for the sessions when registering the edge to the tunnel server